packages/Search/examples/distributed_tree_driver/distributed_tree_driver.cpp

```diff
@@ -0,0 +1,296 @@
+        std::uniform_real_distribution<double> distribution( -a, +a );
+        std::default_random_engine generator;
+        auto random = [&distribution, &generator]() {
+            return distribution( generator );
+        };
+
+        double offset_x = 0.;
+        double offset_y = 0.;
+        double offset_z = 0.;
+        // Change the geometry of the problem. In 1D, all the point clouds are
+        // aligned on a line. In 2D, the point clouds create a board and in 3D,
+        // they create a box.
+        switch ( dim )
+        {
+        case 1:
+        {
+            offset_x = 2. * ( 1. - overlap ) * a * rank;
+
+            break;
+        }
+        case 2:
+        {
+            int n_procs = Teuchos::size( *comm );
+            int i_max = std::ceil( std::sqrt( n_procs ) );
+            int i = rank % i_max;
+            int j = rank / i_max;
+            offset_x = 2. * ( 1. - overlap ) * a * i;
+            offset_y = 2. * ( 1. - overlap ) * a * j;
+
+            break;
+        }
+        case 3:
+        {
+            int n_procs = Teuchos::size( *comm );
+            int i_max = std::ceil( std::cbrt( n_procs ) );
+            int j_max = i_max;
+            int i = rank % i_max;
+            int j = ( rank / i_max ) % j_max;
+            int k = rank / ( i_max * j_max );
+            offset_x = 2. * ( 1. - overlap ) * a * i;
+            offset_y = 2. * ( 1. - overlap ) * a * j;
+            offset_z = 2. * ( 1. - overlap ) * a * k;
+
+            break;
+        }
+        default:
+        {
+            throw std::runtime_error( "dim should be 1, 2, or 3" );
+        }
+        }
+
+        for ( int i = 0; i < n; ++i )
+            random_points_host( i ) = {{offset_x + random(),
+                                        offset_y + random(),
+                                        offset_z + random()}};
+        Kokkos::deep_copy( random_points, random_points_host );
+    }
+
+    Kokkos::View<DataTransferKit::Box *, DeviceType> bounding_boxes(
+        Kokkos::ViewAllocateWithoutInitializing( "bounding_boxes" ), n_values );
+    Kokkos::parallel_for( "bvh_driver:construct_bounding_boxes",
+                          Kokkos::RangePolicy<ExecutionSpace>( 0, n_values ),
+                          KOKKOS_LAMBDA( int i ) {
+                              double const x = random_points( i )[0];
+                              double const y = random_points( i )[1];
+                              double const z = random_points( i )[2];
+                              bounding_boxes( i ) = {
+                                  {{x - 1., y - 1., z - 1.}},
+                                  {{x + 1., y + 1., z + 1.}}};
+                          } );
+    Kokkos::fence();
+
+    auto construction = time_monitor.getNewTimer( "construction" );
+    construction->start();
+    DataTransferKit::DistributedSearchTree<DeviceType> distributed_tree(
+        comm, bounding_boxes );
+    construction->stop();
```

Comment:
Is it possible to get an intermediate output to stdout to signify that tree has been constructed at that point?
I am not so keen on having to wait execution all the way to the very end of the driver to get any timing.

Reply:
This shows as outdated because MPI barriers were added in the meantime.  Please do not overlook.